pycqed/measurement/calibration_points.py

```diff
@@ -72,26 +75,44 @@ def get_states(self, qb_names=None):
         return {qbn: [s[self.qb_names.index(qbn)] for s in self.states]
                 for qbn in qb_names}
 
-    def get_indices(self, qb_names=None):
+    def get_indices(self, qb_names=None, prep_params=None):
         """
         Get calibration indices
         Args:
             qb_names: qubit name or list of qubit names to retrieve
                 the indices of. Defaults to all.
+            prep_params: QuDev_transmon preparation_params attribute
 
         Returns: dict where keys are qb_names and values dict of {state: ind}
 
         """
+        if prep_params is None:
+            prep_params = self.prep_params
+        if prep_params is None:
+            prep_params = {}
+        prep_type = prep_params.get('preparation_type', 'wait')
```

Comment:
I think these would be the lines I would keep and I would remove the state in self. But maybe I'm just missing something about the usecase